internal/graph/check.go

```diff
@@ -688,22 +735,37 @@ func (c *LocalChecker) checkUsersetSlowPath(ctx context.Context, iter storage.Tu
 		}
 	}
 
-	if len(handlers) == 0 && errs != nil {
-		telemetry.TraceError(span, errs)
-		return nil, errs
-	}
-
 	resp, err := union(ctx, c.concurrencyLimit, handlers...)
 	if err != nil {
 		telemetry.TraceError(span, err)
-		return nil, errors.Join(errs, err)
+		return nil, err
 	}
 
 	resp.GetResolutionMetadata().DatastoreQueryCount += response.GetResolutionMetadata().DatastoreQueryCount
 
 	return resp, nil
 }
 
+func TupleKeyConditionalFilter(ctx context.Context, reqCtx *structpb.Struct, typesys *typesystem.TypeSystem) storage.TupleKeyConditionFilterFunc {
```

Comment:
Maybe this should be private (only meant to be used in Check for now) and perhaps it should be called `buildTupleKeyConditionFilter`?